zeppelin-web/test/spec/controllers/paragraph.js

```diff
@@ -1,19 +1,82 @@
 'use strict';
 
-describe('Controller: ParagraphCtrl', function () {
+describe('Controller: ParagraphCtrl', function() {
 
-  // load the controller's module
   beforeEach(module('zeppelinWebApp'));
 
-  var ParagraphCtrl, scope;
+  var ParagraphCtrl;
+  var scope;
+  var websocketMsgSrvMock = {};
+  var paragraphMock = {
+    config: {}
+  };
 
-  // Initialize the controller and a mock scope
-  beforeEach(inject(function ($controller, $rootScope) {
+  beforeEach(inject(function($controller, $rootScope) {
     scope = $rootScope.$new();
     ParagraphCtrl = $controller('ParagraphCtrl', {
-      $scope: scope
+      $scope: scope,
+      websocketMsgSrv: websocketMsgSrvMock,
+      $element: {}
     });
+    scope.init(paragraphMock);
   }));
 
-  //Write test to test ParagraphCtrl
+  var functions = ['isRunning', 'getIframeDimensions', 'cancelParagraph', 'runParagraph', 'saveParagraph',
+    'moveUp', 'moveDown', 'insertNew', 'removeParagraph', 'toggleEditor', 'closeEditor', 'openEditor',
+    'closeTable', 'openTable', 'showTitle', 'hideTitle', 'setTitle', 'showLineNumbers', 'hideLineNumbers',
+    'changeColWidth', 'columnWidthClass', 'toggleGraphOption', 'toggleOutput', 'loadForm',
+    'aceChanged', 'aceLoaded', 'getEditorValue', 'getProgress', 'getExecutionTime', 'isResultOutdated',
+    'getResultType', 'loadTableData', 'setGraphMode', 'isGraphMode', 'onGraphOptionChange',
+    'removeGraphOptionKeys', 'removeGraphOptionValues', 'removeGraphOptionGroups', 'setGraphOptionValueAggr',
+    'removeScatterOptionXaxis', 'removeScatterOptionYaxis', 'removeScatterOptionGroup',
+    'removeScatterOptionSize'];
+
+  functions.forEach(function(fn) {
+    it('check for scope functions to be defined : ' + fn, function() {
+      expect(scope[fn]).toBeDefined();
```

Comment:
It is good test functionality of the function, rather only checking is function is define. 

In paragraph ctrl you will find lot functions and checking them all  will not be easy. 
There for you can test for use cases